server/controllers/user.js

```diff
@@ -0,0 +1,168 @@
+require('dotenv').load();
+const User = require('./../models/user');
+const nJwt = require('njwt');
+
+const secret = process.env.SECRET;
+
+const userCntrl = {
+  authenticate: (req, res) => {
+    // check header or url parameters or post parameters for token
+    const token = req.body.token || req.query.token || req.headers['x-access-token'];
+    if (token) {
+      let verified = nJwt.verify(token, secret);
+      if (verified === { JwtParseError: 'Jwt cannot be parsed' }) {
+        verified = { message: 'Unauthorized User!' };
+        return verified;
+      }
+      return verified
+    }
+    const verified = { message: 'Unauthorized User!' };
+    return verified;
+  },
+  createUser: (req, res) => {
+    const token = userCntrl.authenticate(req, res);
+    if (token.message === 'Unauthorized User!') {
+      return res.status(400).json({ message: 'Unauthorized User!' });
+    }
+    const permissions = token.body.permissions;
+
+    User.find({ email: req.body.email }, (err, users) => {
+      if (!users.length) {
+        const user = new User(); // create a new instance of the User models
+
+        user.userName = req.body.userName;
+        user.name.first = req.body.firstName;
+        user.name.last = req.body.lastName;
+        user.email = req.body.email;
+        user.password = user.generateHash(req.body.password);
+        if (permissions === 'Admin') {
+          user.role = req.body.role || 'User';
+        } else {
+          user.role = 'User';
+        }
+
+        // save the user and check for errors
+        user.save(() => {
+          if (err) {
+            res.status(400).json({
+              message: 'Something went wrong',
+              err,
+            });
+          } else {
+            res.status(200).json({ message: 'User Created', user });
+          }
+        });
+      } else {
+        res.status(400).json({ message: 'User already exists' });
+      }
+    });
+  },
+  all: (req, res) => {
+    User.find((err, users) => {
+      if (err) {
+        return res.send(err);
+      }
+      return res.json({
+        message: 'all users',
+        status: 200,
+        users,
+      });
+    });
+  },
+
+  getSpecificUser: (req, res) => {
+    User.findById(req.params.user_id, (err, user) => {
+      if (err) {
+        res.status(400).json({ err });
```

Comment:
**Consistent returns**
You need to be consistent in what the API returns. Best practice is to just return the error object.